app/models/user.js

```diff
@@ -359,14 +372,28 @@ exports.addModel = function(dbAdapter) {
       hasChanges = true
     }
 
+    if (params.hasOwnProperty('frontendPreferences')) {
+      this.frontendPreferences = params.frontendPreferences
+      hasChanges = true
+    }
+
     if (hasChanges) {
       this.updatedAt = new Date().getTime()
 
+      // Validate maximum preferences store size
```

Comment:
Why are you validating it here, not in the `if` (line 376), as other properties do? In here it will be validated every time even if the client doesn't change it.